packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx

```diff
@@ -144,7 +144,13 @@ const MessageBubble = (props: ChatMessageComponentAsMessageBubbleProps): JSX.Ele
           onTouchStart={() => setWasInteractionByTouch(true)}
           onPointerDown={() => setWasInteractionByTouch(false)}
           onKeyDown={() => setWasInteractionByTouch(false)}
-          onClick={() => wasInteractionByTouch && setChatMessageActionFlyoutTarget(messageRef)}
+          onClick={() => {
+            if (!wasInteractionByTouch) {
+              return;
+            }
+            props.onActionButtonClick(message, setMessageReadBy);
+            setChatMessageActionFlyoutTarget(messageRef);
```

Comment:
I had to remember what was happening here... can you do better than me and add a comment haha?
```ts
// If the message was touched via touch we immediately open the menu flyout (when using mouse the 3-dot menu that appears on hover  must be clicked to open the flyout).
// In doing so here we set the target of the flyout to be the message and not the 3-dot menu button to position the flyout correctly.
```